structure/homology_modeling_functions.py

```diff
@@ -268,9 +268,11 @@ def pdb_array_creator(self, structure=None, filename=None):
                             gn = gn+'0'
                         if gn[0]=='-':
                             gn = gn[1:]+'1'
-                        # Exception for 3PBL 331, gn get's assigned wrong
+                        # Exceptions
```

Comment:
A cleaner way could be to set a variable to "True" in order to keep track of cases that need to be handled alternatively. The current implementation  is somewhat abusing the try/except system.


```
for residue in pdb_struct[pref_chain]:
    alternative = False 
    ......
    if structure.pdb_code.index=='3PBL' and residue.get_id()[1]==331:
        alternative = True
     ...
    if (alternative):
           gn_array.append(str(residue.get_id()[1]))
           residue_array.append(residue.get_list())
```